yaksh/test_models.py

```diff
@@ -1084,6 +1095,56 @@ def setUpClass(self):
         )
         self.mcc_based_testcase.save()
 
+        # create answerpaper2
+        self.answerpaper2 = AnswerPaper(
+            user=self.user,
+            question_paper=self.question_paper2,
+            start_time=self.start_time,
+            end_time=self.end_time,
+            user_ip=self.ip,
+            course=self.course
+        )
+
+        self.attempted_papers2 = AnswerPaper.objects.filter(
+            question_paper=self.question_paper2,
+            user=self.user
+        )
+
+        self.question_paper2.fixed_questions.add(*self.questions2)
```

Comment:
Why not first create question paper completely and then create answer paper?

Reply:
What difference does that make?

Reply:
It makes a difference.
You are creating question paper incompletely then creating answer paper with that incomplete question paper and saving it then,
you add questions to question paper.
Then add those questions to answer paper n save it!
They are behaving independently when they are related.
Also it increases complexity unnecessary.
Simply create a question paper completely, then u have make_answerpaper method of question paper to create answer paper. 
So simple and clear!